gax-httpjson/src/main/java/com/google/api/gax/httpjson/ApiMessageOperationTransformers.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google LLC nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.google.api.gax.httpjson;
+
+import com.google.api.core.ApiFunction;
+import com.google.api.core.BetaApi;
+import com.google.api.gax.longrunning.OperationSnapshot;
+import com.google.api.gax.rpc.ApiExceptionFactory;
+import com.google.api.gax.rpc.StatusCode.Code;
+
+/**
+ * Transformers from OperationSnapshot wrappers to the underlying native ApiMessage objects. Public
+ * for technical reasons; intended for use by generated code.
+ */
+@BetaApi("The surface for use by generated code is not stable yet and may change in the future.")
+public class ApiMessageOperationTransformers {
+  private ApiMessageOperationTransformers() {}
+
+  public static class ResponseTransformer<ResponseT extends ApiMessage>
+      implements ApiFunction<OperationSnapshot, ResponseT> {
+    private final Class<ResponseT> responseTClass;
+
+    private ResponseTransformer(Class<ResponseT> responseTClass) {
+      this.responseTClass = responseTClass;
+    }
+
+    /** Unwraps an OperationSnapshot and returns the contained method response message. */
+    @SuppressWarnings("unchecked")
+    public ResponseT apply(OperationSnapshot operationSnapshot) {
+      if (!operationSnapshot.getErrorCode().getCode().equals(Code.OK)) {
```

Comment:
Same issue applies to `ProtoOperationTransformers`. It looks like we are interested if the call was successful or no. This treats only `200` as success, but any `2xx` code can be considered success [HTTP Status Codes](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#2xx_Success). 

Though, this is unlikely that we can get anything but `200` here for success in practical cases (maybe `201` also). Please add at least a comment about "potential need to handle other error codes". If that ever happens (other than 200 success status code returned) it may save time for those who will be fixing it.

Reply:
TIL
See the new diff in HttpJsonStatusCode.java to squash all 2xx codes into the OK 200 code.

Reply:
Please add a comment here about potential need to handle more than only 200.